libs/api/domains/license-service/src/lib/mappers/pCardMapper.ts

```diff
@@ -1,94 +1,120 @@
 import isAfter from 'date-fns/isAfter'
 import { Locale } from '@island.is/shared/types'
 import {
-  GenericLicenseDataField,
-  GenericLicenseDataFieldType,
-  GenericLicenseLabels,
-  GenericLicenseMapper,
-  GenericUserLicensePayload,
-} from '../licenceService.type'
-import { DEFAULT_LICENSE_ID } from '../licenseService.constants'
-import { getLabel } from '../utils/translations'
+  DEFAULT_LICENSE_ID,
+  LICENSE_NAMESPACE,
+} from '../licenseService.constants'
 import { Injectable } from '@nestjs/common'
 import { Staediskortamal } from '@island.is/clients/p-card'
 import { isDefined } from '@island.is/shared/utils'
 import { format } from 'kennitala'
+import { m } from '../messages'
+import { IntlService } from '@island.is/cms-translations'
+import { expiryTag } from '../utils/expiryTag'
+import { formatDate } from '../utils'
+import {
+  GenericLicenseDataFieldType,
+  GenericLicenseMappedPayloadResponse,
+  GenericLicenseMapper,
+} from '../licenceService.type'
+import { GenericLicenseDataField } from '../dto/GenericLicenseDataField.dto'
 
 @Injectable()
 export class PCardPayloadMapper implements GenericLicenseMapper {
-  parsePayload(
+  constructor(private readonly intlService: IntlService) {}
+  async parsePayload(
     payload: Array<unknown>,
     locale: Locale = 'is',
-    labels?: GenericLicenseLabels,
-  ): Array<GenericUserLicensePayload> {
-    if (!payload) return []
+  ): Promise<Array<GenericLicenseMappedPayloadResponse>> {
+    if (!payload) return Promise.resolve([])
```

Comment:
Should this warn or log some error, or some indication that the payload is empty?

Reply:
Not really an error